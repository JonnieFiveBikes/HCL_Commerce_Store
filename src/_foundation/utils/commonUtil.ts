/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */

const commonUtil = {
  /**
   * Perform link action accordingly
   * @param link the link contains action instruction
   */
  parseConentUrl: (link: string): any => {
    /**
     * case 1: add to cart
     *  OrderItemAdd?partNumber=LR-DECO-0003-0001&URL=OrderCalculate?URL=OrderItemDisplay&calculationUsageId=-1&quantity=1&catalogId=#catalogId#&storeId=#storeId#
     *
     */
    if (link.indexOf("OrderItemAdd?") === 0) {
      return {
        action: "addToCart",
        partnumber: [
          link.replace("OrderItemAdd?", "").split("&")[0].split("=")[1],
        ],
      };
    } else if (link.indexOf("InterestItemAdd") === 0) {
      /**
       * case 2: add to wishlist
       *  InterestItemAdd?partNumber=AuroraWMDRS-001&URL=InterestItemDisplay&catalogId=#catalogId#&storeId=#storeId#
       *
       */
      return {
        action: "addToWishlist",
        partnumber: [
          link.replace("InterestItemAdd?", "").split("&")[0].split("=")[1],
        ],
      };
    } else if (
      /**
       * case 3: "/home"
       *
       */
      !link.includes("?") &&
      !link.includes("=") &&
      link.split("/").length > 1
    ) {
      link = link.substring(link.indexOf("/"));
      return {
        action: "navigate",
        toLink: link,
      };
    } else if (link.startsWith("http://") || link.startsWith("https://")) {
      /**
       * case 4: external link
       *
       */
      window.open(link, "_blank");
    } else {
      return {
        action: "navigate",
        toLink: link,
      };
    }
  },
};

export default commonUtil;
