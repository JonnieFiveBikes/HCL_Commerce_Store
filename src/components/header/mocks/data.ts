/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */
import { OrderReducerState } from "../../../redux/reducers";

interface MiniCartTestState {
  order: OrderReducerState;
}
const initStatesBase: OrderReducerState = {
  cart: null,
  numItems: 0,
  orderItems: [],
  catentries: null,
  isCheckoutDisabled: false,
  shipInfos: null,
  shipModes: [],
  payMethods: [],
  isRecurringOrderDisabled: false,
};

export const ONE_ITEM = 1;
export const ORDERITEM_1 = {
  unitUom: "C62",
  shippingChargeCurrency: "USD",
  lastUpdateDate: "2020-08-31T21:00:19.591Z",
  expectedShipDate: "2020-08-31T21:00:19.747Z",
  description: "Mail",
  language: "-1",
  salesTax: "0.00000",
  xitem_isPersonalAddressesAllowedForShipping: "true",
  correlationGroup: "10065",
  shippingTax: "0.00000",
  orderItemStatus: "P",
  offerID: "4000000000000008037",
  currency: "USD",
  shippingCharge: "0.00000",
  orderItemPrice: "749.99000",
  shipModeLanguage: "-1",
  createDate: "2020-08-31T21:00:19.576Z",
  unitPrice: "749.99000",
  salesTaxCurrency: "USD",
  quantity: "1.0",
  shipModeCode: "Mail",
  productId: "14019",
  shipModeDescription: "Mail",
  orderItemId: "10065",
  fulfillmentCenterId: "13001",
  fulfillmentCenterName: "Emerald",
  shipModeId: "14001",
  isExpedited: "false",
  orderItemFulfillmentStatus: "Unreleased",
  shippingTaxCurrency: "USD",
  carrier: "Mail",
  UOM: "C62",
  freeGift: "false",
  unitQuantity: "1.0",
  contractId: "-11005",
  adjustment: [
    {
      amount: "-25.00000",
      code: "Save $25 on all orders over $200 USD",
      displayLevel: "Order",
      usage: "Discount",
      descriptionLanguage: "-1",
      description: "Save $25 on all orders over $200 USD",
      language: "-1",
      currency: "USD",
    },
  ],
  partNumber: "LR-FNTR-0001-0001",
  totalAdjustment: {
    currency: "USD",
    value: "-25.00000",
  },
  orderItemInventoryStatus: "Available",
  name: "Wooden Angled Chair",
  seo: { href: "/wooden-angled-chair-lr-fntr-0001-0001" },
  thumbnail:
    "/hclstore/EmeraldCAS//images/catalog/livingroom/furniture/chair1_a1_350.jpg",
  attributes: [
    {
      identifier: "Color",
      usage: "Defining",
      values: [
        {
          identifier: "blonde",
          sequence: "1.0",
          unitOfMeasure: "one",
          unitID: "C62",
          id: "7000000000000003005",
          image1: "/images/catalog/swatches/sw_blonde.png",
          value: "Blonde",
          image1path:
            "/hclstore/EmeraldCAS//images/catalog/swatches/sw_blonde.png",
        },
      ],
      displayable: "true",
      merchandisable: "true",
      searchable: "true",
      sequence: "0.0",
      name: "Color",
      facetable: "true",
      id: "7000000000000000501",
      comparable: "true",
      key: "color",
      swatchable: "false",
    },
  ],
};
export const CART_1 = {
  totalShippingCharge: "0.00000",
  resourceId:
    "https://localhost/wcs/resources/store/11/cart/@self?pageNumber=1&pageSize=3&sortOrder=DESC&sortOrderItemBy=orderItemID&langId=-1",
  orderId: "13007517",
  lastUpdateDate: "2020-08-31T21:00:19.578Z",
  channel: {
    channelIdentifer: {
      channelName: "Telesales",
      uniqueID: "-4",
    },
    userData: null,
    description: {
      language: "-1",
      value: "Used when operations performed by a telesales representative.",
    },
  },
  orderStatus: "P",
  x_isPurchaseOrderNumberRequired: "false",
  totalShippingChargeCurrency: "USD",
  grandTotalCurrency: "USD",
  buyerId: "13",
  recordSetCount: "1",
  x_isPersonalAddressesAllowedForShipping: "true",
  storeNameIdentifier: "Emerald",
  totalProductPriceCurrency: "USD",
  totalProductPrice: "749.99000",
  locked: "false",
  recordSetComplete: "true",
  totalAdjustmentCurrency: "USD",
  totalSalesTaxCurrency: "USD",
  totalSalesTax: "0.00000",
  grandTotal: "724.99000",
  storeUniqueID: "11",
  recordSetStartNumber: "0",
  resourceName: "cart",
  recordSetTotal: "1",
  shipAsComplete: "true",
  x_trackingIds: "",
  totalShippingTax: "0.00000",
  totalShippingTaxCurrency: "USD",
  prepareIndicator: "false",
  adjustment: [
    {
      amount: "-25.00000",
      code: "Save $25 on all orders over $200 USD",
      displayLevel: "Order",
      usage: "Discount",
      xadju_calUsageId: "-1",
      descriptionLanguage: "-1",
      description: "Save $25 on all orders over $200 USD",
      language: "-1",
      currency: "USD",
    },
  ],
  totalAdjustment: "-25.00000",
};

export const initStatesEmptyCart: MiniCartTestState = {
  order: {
    ...initStatesBase,
  },
};
export const initStatesOneItem: MiniCartTestState = {
  order: {
    ...initStatesBase,
    cart: CART_1,
    numItems: ONE_ITEM,
    orderItems: [ORDERITEM_1],
  },
};
