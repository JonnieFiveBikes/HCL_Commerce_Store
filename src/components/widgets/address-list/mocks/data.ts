/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */
export const cidIValue = "address-list";
export const addressList = [
  {
    firstName: "John",
    lastName: "Doe",
    zipCode: "12345",
    country: "US",
    email1: "billing@mailinator.com",
    city: "Schenectady",
    nickName: "billing-joe",
    addressType: "Billing",
    state: "NY",
    addressLine: ["1 Sesame Street", "", ""],
    addressId: "3074457354534345997",
    primary: "false",
  },
  {
    lastName: "Smith",
    zipCode: "54321",
    country: "US",
    city: "NYC",
    nickName: "shipping-jane",
    addressType: "Shipping",
    addressLine: ["1 Main Street", "Unit #2", ""],
    addressId: "3074457354535646440",
    phone1: "4161234567",
    firstName: "Jane",
    email1: "shipping@mailinator.com",
    state: "NV",
    primary: "false",
  },
];
