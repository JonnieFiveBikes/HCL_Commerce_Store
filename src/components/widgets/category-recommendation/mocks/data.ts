/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */
import { RootReducerState } from "../../../../redux/reducers";

export const initStates: RootReducerState = {
  account: {
    orders: null,
    address: null,
  },
  api: {
    apiFlowList: [],
  },
  catalog: {
    productList: [],
    productListTotal: -1,
    priceMode: "1",
    facets: null,
    facetPrice: null,
    selectedFacets: {},
    selectedFacetLimits: [],
    selectedFacetPrices: {
      min: -1,
      max: -1,
    },
    selectedPageOffset: 1,
    selectedSortOption: "0",
    breadcrumbs: [],
  },
  user: {
    WCToken: "",
    WCTrustedToken: "",
    lastUpdated: 1111,
  },
  order: {
    cart: null,
    numItems: 0,
    orderItems: [],
    catentries: null,
    isCheckoutDisabled: false,
    shipInfos: null,
    shipModes: [],
    payMethods: [],
    isRecurringOrderDisabled: false,
  },
  error: {
    errorKey: null,
    errorCode: null,
    errorTitleKey: null,
    errorMsgKey: null,
    handled: null,
    errorMessage: "",
    errorParameters: "",
  },
  seo: {},
  contract: {
    "-11005": "-11005",
  },
  search: {
    keywords: [],
  },
  context: {
    globalization: {
      preferredCurrency: "USD",
      languageId: -1,
      currency: "USD",
      preferredLanguageId: -1,
    },
    catalog: {
      catalogId: 11501,
      masterCatalog: false,
    },
    entitlement: {
      eligibleTradingAgreementIds: ["-11005"],
      hostingContractId: "-11004",
      currentTradingAgreementIds: ["-11005"],
      activeOrganizationId: "-2000",
      sessionTradingAgreementIds: null,
    },
    isPartiallyAuthenticated: false,
    basicInfo: {
      runAsId: -1002,
      callerId: -1002,
      registerType: "G",
      storeId: 11,
      channelId: -4,
    },
  },
  organization: {
    entitledOrganizations: [],
    organizationDetails: {},
    accountCheck: true,
  },
  success: { key: "" },
  confirmation: { key: "", title: "" },
  recurringOrder: {
    resultList: [],
  },
  orderDetails: {},
  site: {
    currentSite: {
      storeName: "Emerald",
      transactionContext: "/wcs/resources",
      searchContext: "/search/resources",
      storeID: "11",
      catalogStoreID: "12001",
      catalogID: "11501",
      defaultCurrencyID: "USD",
      defaultLanguageID: "-1",
      storeType: "MHS",
      isB2B: false,
      storeCfg: {
        identifier: "Emerald",
        locationInfo: [
          {
            address: {
              country: "CA",
              city: "Clothesville",
              zipOrPostalCode: "Lxx 1xx",
              addressLine: ["12xx Martindale Avenue", "Suite 9xx", null],
              stateOrProvinceName: "ON",
            },
            languageId: "-1",
            emailAddress1: "info@Emerald.xxx",
            telephone1: "1-800-555-1234",
            contactInfoIdentifier: {
              contactPersonNickName: "Emerald location_en_US",
              contactInfoId: "13002",
            },
            fax1: "1-800-555-4321",
            contactPersonBusinessTitle: "Emerald",
          },
        ],
        relatedStores: [
          {
            relationshipType: "-34",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-34",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-33",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-33",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-32",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-32",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-31",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-31",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-29",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-29",
            relatedStoreId: "12001",
            relationshipSequence: "2.0",
            state: "1",
          },
          {
            relationshipType: "-29",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-26",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-26",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-25",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-25",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-24",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-24",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-23",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-23",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-22",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-22",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-21",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-21",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-20",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-20",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-19",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-19",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-18",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-18",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-17",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-17",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-16",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-16",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-14",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-14",
            relatedStoreId: "12001",
            relationshipSequence: "2.0",
            state: "1",
          },
          {
            relationshipType: "-14",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-11",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-11",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-10",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-10",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-9",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-9",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-7",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-7",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-5",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-5",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-4",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-4",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-3",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-3",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-2",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-2",
            relatedStoreId: "12501",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-1",
            relatedStoreId: "11",
            relationshipSequence: "0.0",
            state: "1",
          },
          {
            relationshipType: "-1",
            relatedStoreId: "12001",
            relationshipSequence: "1.0",
            state: "1",
          },
          {
            relationshipType: "-1",
            relatedStoreId: "12501",
            relationshipSequence: "2.0",
            state: "1",
          },
        ],
        storeType: "MHS",
        userData: {
          "wc.search.priceMode.compatiblePriceIndex": "1.0",
          headlessStore: "true",
          "headlessStore.storeContextRoot": "/Emerald",
        },
        contactInfo: [
          {
            address: {
              country: "CA",
              city: "Clothesville",
              zipOrPostalCode: "Lxx 1xx",
              addressLine: ["12xx Martindale Avenue", "Suite 9xx", null],
              stateOrProvinceName: "ON",
            },
            languageId: "-1",
            emailAddress1: "info@Emerald.xxx",
            telephone1: "1-800-555-1234",
            contactInfoIdentifier: {
              contactPersonNickName: "Emerald contact_en_US",
              contactPersonId: "-11000",
              contactInfoId: "13001",
            },
            fax1: "1-800-555-4321",
            contactPersonBusinessTitle: "Emerald",
          },
        ],
        description: [
          {
            displayName: "Emerald",
            languageId: "-1",
            description: "Commerce Model Store entity",
          },
        ],
        defaultCatalog: [
          {
            userData: {
              userDataField: [],
            },
            catalogIdentifier: {
              externalIdentifier: null,
              uniqueID: "11501",
            },
            storeIdentifier: {
              externalIdentifier: null,
              uniqueID: "11",
            },
            uniqueID: "11001",
          },
        ],
        storeId: "11",
        ownerId: "-11000",
        supportedCurrencies: {
          supportedCurrencies: ["USD"],
          defaultCurrency: "USD",
        },
        supportedLanguages: {
          supportedLanguages: ["-1"],
          defaultLanguageId: "-1",
        },
        state: "open",
      },
      supportedLanguages: ["-1"],
      supportedCurrencies: ["USD"],
      inventorySystem: -2,
    },
  },
};

export const cidIValue = "home-home-category-rec";
export const eSpotIValue = {
  eSpotName: "Home_CategoryRec",
};
export const page = {
  pageName: "home",
};
