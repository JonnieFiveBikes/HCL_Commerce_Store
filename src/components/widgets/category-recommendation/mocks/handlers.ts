/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */
import { rest } from "msw";
import * as homeCatESpot from "../../../../../mocks/wcs/resources/store/__/espot/data/HomeCategoryRec.mock.json";
import * as categoryRecs from "../../../../../mocks/search/resources/api/v2/categories/data/10516.10501.mock.json";
export const handlers = [
  // Handles a GET /user request
  rest.get(
    "http://localhost/wcs/resources/store/:storeId/espot/Home_CategoryRec",
    (req, res, ctx) => {
      return res(ctx.json(homeCatESpot));
    }
  ),
  rest.get(
    "http://localhost/search/resources/api/v2/categories",
    (req, res, ctx) => {
      return res(ctx.json(categoryRecs));
    }
  ),
];
