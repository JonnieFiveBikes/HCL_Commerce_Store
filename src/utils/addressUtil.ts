/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */
//Custom libraries
import { REG_EX } from "../constants/common";

const addressUtil = {
  optionalAddressFields: ["addressLine2", "phone1"],
  addressFormFields: [
    "firstName",
    "lastName",
    "addressLine1",
    "addressLine2",
    "city",
    "country",
    "state",
    "zipCode",
    "phone1",
    "nickName",
    "email1",
    "addressType",
  ],

  validateAddressForm: (formData: any, edit?: boolean) => {
    const editVal = edit ? edit : false;

    for (let key in formData) {
      if (!addressUtil.optionalAddressFields.includes(key)) {
        if (formData[key] !== undefined && formData[key].trim() === "") {
          return false;
        }
      }
    }

    if (!addressUtil.validatePhoneNumber(formData.phone1)) {
      return false;
    }
    if (!addressUtil.validateEmail(formData.email1)) {
      return false;
    }

    if (!editVal && !addressUtil.validateNickName(formData.nickName)) {
      return false;
    }

    return true;
  },

  validatePhoneNumber: (phoneNumber: string) => {
    const PHONE = REG_EX.PHONE;
    return (
      phoneNumber === undefined ||
      phoneNumber.trim() === "" ||
      PHONE.test(phoneNumber.trim())
    );
  },

  validateEmail: (email: string) => {
    const EMAIL = REG_EX.EMAIL;
    return email === undefined || email.trim() === "" || EMAIL.test(email);
  },

  validateNickName: (nickName: string) => {
    const NICKNAME_ALPHA_NUMERIC_SPECIAL_CHAR =
      REG_EX.NICKNAME_ALPHA_NUMERIC_SPECIAL_CHAR;
    return (
      nickName === undefined ||
      nickName.trim() === "" ||
      NICKNAME_ALPHA_NUMERIC_SPECIAL_CHAR.test(nickName)
    );
  },

  removeLeadingTrailingWhiteSpace: (formData: any): any => {
    let result: any = {};
    // remove leading and trailing white space from all form input fields.
    if (formData !== undefined && formData !== null) {
      Object.keys(formData).forEach((key) => {
        let value = formData[key].trim();
        result[key] = value;
      });
    }
    return result;
  },

  removeIgnorableAddressFormFields: (formData: any): any => {
    let result: any = { ...formData };
    for (let key in result) {
      if (!addressUtil.addressFormFields.includes(key)) {
        delete result[key];
      }
    }
    return result;
  },
};

export default addressUtil;
