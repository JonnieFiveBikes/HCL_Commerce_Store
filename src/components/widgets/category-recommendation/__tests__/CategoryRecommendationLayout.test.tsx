/*
 *==================================================
 * Licensed Materials - Property of HCL Technologies
 *
 * HCL Commerce
 *
 * (C) Copyright HCL Technologies Limited 2020
 *
 *==================================================
 */

import React from "react";
import { render, waitForElement } from "../../../../testing/utils/test-utils";
import { CategoryRecommendationLayout } from "../";
import { initStates, cidIValue, eSpotIValue, page } from "../mocks/data";
import { server } from "../../../../testing/mocks/server";
import { handlers } from "../mocks/handlers";

jest.mock(
  "../../category-card/CategoryCardLayout",
  () => ({ renderingContext }) => (
    <div data-testid={`category-${renderingContext.id}`} />
  )
);
describe("Category Recommendation Layout", () => {
  test("renders empty Category Recommendation when eSpot not found", async () => {
    const { container, queryByTestId } = render(
      <CategoryRecommendationLayout
        cid={cidIValue}
        eSpot={eSpotIValue}
        page={page}
      />,
      {
        initialState: initStates as any,
      }
    );
    const element = await waitForElement(() =>
      container.querySelector(".vertical-padding-2")
    );
    expect(element).toBeTruthy();
    expect(element).toMatchSnapshot();
    const cat = await waitForElement(() =>
      queryByTestId("category-Kitchen")
    ).catch((e) => null);
    expect(cat).toBeFalsy();
  });

  test("renders Category Recommendation with eSpot and category data", async () => {
    server.use(...handlers);
    const { queryByTestId } = render(
      <CategoryRecommendationLayout
        cid={cidIValue}
        eSpot={eSpotIValue}
        page={page}
      />,
      {
        initialState: initStates as any,
      }
    );
    const cat = await waitForElement(() => queryByTestId("category-Kitchen"));
    expect(cat).toBeTruthy();
    expect(cat).toMatchSnapshot();
  });
});
