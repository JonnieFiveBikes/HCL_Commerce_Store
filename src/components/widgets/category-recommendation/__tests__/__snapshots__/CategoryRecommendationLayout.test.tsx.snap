// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Recommendation Layout renders Category Recommendation with eSpot and category data 1`] = `
<div
  data-testid="category-Kitchen"
/>
`;

exports[`Category Recommendation Layout renders empty Category Recommendation when eSpot not found 1`] = `
<div
  class="MuiGrid-root sc-ezrdKe dsbMot vertical-padding-2 MuiGrid-container MuiGrid-spacing-xs-2"
/>
`;
